### PR TITLE
fix(headless-fixes): remove InputSystem and NuiManager from headlesses

### LINKED ISSUE #3793: Seemingly benign NPE from initializing InputSystem for a headless server
### What you were trying to do

Test stuff! Clear logs are happy logs

### What actually happened

A `NullPointerException` got logged breaking the perfect zen moment of staring at the logs :-( Actually, not really new, been seeing this for months if not longer.

Chances are some code gets initialized that makes no sense in a headless situation. May also want to check the [ModuleTestingEnvironment](https://github.com/Terasology/ModuleTestingEnvironment) starting a test since it uses both a headless server _and_ headless clients.

Need to find where it gets logged then see if it makes sense to do a test of the network mode or something to only attempt that initialization where it makes sense, trying a little detective work in case it does actually result in something undesirable.

### How to reproduce

* Run a headless server, seemingly any config
* Look through the logging, notice NPE during startup (after discovering modules, before activating them)
* Observe the server still working just fine, so it doesn't _seem_ to be very bad ..

### Log snippet

```
20:27:10.450 [main] INFO  o.t.i18n.TranslationSystemImpl - Discovered engine:menu_ru
20:27:10.754 [main] INFO  o.terasology.engine.TerasologyEngine - Initialization completed in 12.24sec.
20:27:11.500 [main] ERROR o.t.engine.ComponentSystemManager - Failed to initialise system org.terasology.input.InputSystem@374b6e33
java.lang.NullPointerException: null
	at org.terasology.input.InputSystem.initialise(InputSystem.java:124)
	at org.terasology.engine.ComponentSystemManager.initialiseSystem(ComponentSystemManager.java:160)
	at org.terasology.engine.ComponentSystemManager.initialise(ComponentSystemManager.java:144)
	at org.terasology.engine.subsystem.headless.mode.StateHeadlessSetup.init(StateHeadlessSetup.java:104)
	at org.terasology.engine.TerasologyEngine.switchState(TerasologyEngine.java:550)
	at org.terasology.engine.TerasologyEngine.changeState(TerasologyEngine.java:533)
	at org.terasology.engine.TerasologyEngine.run(TerasologyEngine.java:399)
	at org.terasology.engine.Terasology.main(Terasology.java:145)
20:27:12.389 [main] WARN  o.t.engine.internal.TimeBase - Delta too great (13205), capping to 1000
20:27:12.400 [main] INFO  o.t.e.m.loadProcesses.RegisterMods - Activating module: engine:2.2.1-SNAPSHOT
```

### LINKED ISSUE #4481: engine with HeadlessInput fails to initialise for lack of a BindsSubsystem
amongst all the noise on the failing MTE tests, I found this

```
18:01:15.610 [main] ERROR o.t.engine.ComponentSystemManager - Failed to initialise system org.terasology.input.InputSystem@53ed80d3
java.lang.NullPointerException: null
	at org.terasology.input.InputSystem.initialise(InputSystem.java:115)
	at org.terasology.engine.ComponentSystemManager.initialiseSystem(ComponentSystemManager.java:164)
	at org.terasology.engine.ComponentSystemManager.initialise(ComponentSystemManager.java:148)
	at org.terasology.engine.subsystem.headless.mode.StateHeadlessSetup.init(StateHeadlessSetup.java:105)
	at org.terasology.engine.TerasologyEngine.switchState(TerasologyEngine.java:584)
	at org.terasology.engine.TerasologyEngine.changeState(TerasologyEngine.java:567)
	at org.terasology.moduletestingenvironment.ModuleTestingEnvironment.createHost(ModuleTestingEnvironment.java:459)
	at org.terasology.moduletestingenvironment.ModuleTestingEnvironment.setup(ModuleTestingEnvironment.java:168)
```

that Null Pointer Exception is https://github.com/MovingBlocks/Terasology/blob/077020fa31aedf48bf2aad186a6a348c5cbc637e/engine/src/main/java/org/terasology/input/InputSystem.java#L113-L116

Unsure how to proceed here. Add a BindsSubsystem to all the engines with an input subsystem? Or is it a bug for one subsystem to have a dependency on another? Or should it gracefully handle `bindsManager` being null?

I also am confused about how the test runner continues to execute the test here instead of abandoning it, that would have made things clearer.

### PR DESCRIPTION
**Pull request by DarkWeird**
### Contains

Removes InputSystem exception from headless 
   Fixes #4481  
   Fixes #3793
Removes NuiManager from headless. (yahoo ! no more screens and jpgs at headless mode - MTE, Server, headless client)

### How to test
Try game in 4 modes:
1. Server(Headless)
2. Host game(from main menu)
3. Connect to server(can reuse 1)
4. Singleplayer

You should:
0. enter
1. play a bit
2. exit as usually

### CODE REVIEW

File: engine/src/main/java/org/terasology/engine/core/subsystem/headless/renderer/HeadlessWorldRenderer.java [hunk 1]
```diff
@@ -48,8 +47,8 @@
     public HeadlessWorldRenderer(Context context) {
         this.worldProvider = context.get(WorldProvider.class);
         this.chunkProvider = context.get(ChunkProvider.class);
-        LocalPlayerSystem localPlayerSystem = context.get(LocalPlayerSystem.class);
-        localPlayerSystem.setPlayerCamera(noCamera);
+//        LocalPlayerSystem localPlayerSystem = context.get(LocalPlayerSystem.class);
+//        localPlayerSystem.setPlayerCamera(noCamera);
```
Comment (keturn) [inline engine/src/main/java/org/terasology/engine/core/subsystem/headless/renderer/HeadlessWorldRenderer.java:51]: Oh, sure, I guess a headless system wouldn't have a local player, would it?

**Review by keturn — COMMENTED** [review]
I am not awake enough to merge this at the moment, but I am hopeful. It should help the speed and signal/noise ratio in MTE tests a lot.

File: engine/src/main/java/org/terasology/engine/core/subsystem/headless/renderer/HeadlessWorldRenderer.java (hunk 1 above)
Comment (DarkWeird) [reply engine/src/main/java/org/terasology/engine/core/subsystem/headless/renderer/HeadlessWorldRenderer.java:51]: Yeah. It should not.
Mte headless host = networkmode.listen-server
There isLocalClient is false :)

Seems headless version was problem with camera and input system

File: engine/src/main/java/org/terasology/engine/logic/chat/ChatSystem.java [hunk 2]
```diff
@@ -50,7 +50,9 @@
 
     @Override
     public void initialise() {
-        overlay = nuiManager.addOverlay(NotificationOverlay.ASSET_URI, NotificationOverlay.class);
+        if(nuiManager != null) {
```
Comment (jdrueckert) [inline engine/src/main/java/org/terasology/engine/logic/chat/ChatSystem.java:53]: what checkstyle says: whitespace missing

File: engine/src/main/java/org/terasology/engine/logic/console/ConsoleSystem.java [hunk 3]
```diff
@@ -29,16 +29,18 @@
...
     @Override
     public void initialise() {
-        overlay = nuiManager.addOverlay(NotificationOverlay.ASSET_URI, NotificationOverlay.class);
-        console.subscribe((Message message) -> {
-            if (!nuiManager.isOpen("engine:console")) {
-                // make sure the message isn't already shown in the chat overlay
-                if (message.getType() != CoreMessageType.CHAT && message.getType() != CoreMessageType.NOTIFICATION
-                        || !nuiManager.isOpen("engine:chat")) {
-                    overlay.setVisible(true);
+        if(nuiManager != null) {
```
Comment (jdrueckert) [inline engine/src/main/java/org/terasology/engine/logic/console/ConsoleSystem.java:32]: whitespace

File: facades/PC/src/main/java/org/terasology/engine/Terasology.java [hunk 4]
```diff
@@ -270,8 +269,8 @@ private void populateSubsystems(TerasologyEngineBuilder builder) {
         if (isHeadless) {
             builder.add(new HeadlessGraphics())
                     .add(new HeadlessTimer())
-                    .add(new HeadlessAudio())
-                    .add(new HeadlessInput());
+                    .add(new HeadlessAudio());
+//                    .add(new HeadlessInput());
```
Comment (jdrueckert) [inline facades/PC/src/main/java/org/terasology/engine/Terasology.java:273]: can this be removed?

**Review by jdrueckert — COMMENTED** [review]
Unfortunately, I can't properly test this until I'm back from vacation. But I left some code-wise review comments at least.

**Review by jdrueckert — APPROVED** [review]
Tested as suggested in the PR description. LGTM

**Pull request merged**